embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/PendingApiCalls.kt

```diff
@@ -59,17 +72,26 @@ internal class PendingApiCalls(
     }
 
     /**
-     * Returns true if the endpoint's queue has reached its limit.
+     * Returns true if there is al least one pending API call in any non rate limited queue.
      */
-    private fun MutableList<PendingApiCall>.hasReachedLimit(endpoint: Endpoint): Boolean {
-        return this.size >= endpoint.getMaxPendingApiCalls()
+    fun hasPendingApiCallsToSend(): Boolean {
+        return pendingApiCallsMap.entries.any {
+            it.value.hasPendingApiCallsToSend(it.key)
+        }
+    }
+
+    /**
+     * Returns true if the queue has at least one pending API call and the endpoint is not rate limited.
+     */
+    private fun MutableList<PendingApiCall>.hasPendingApiCallsToSend(endpoint: Endpoint): Boolean {
```

Comment:
You can probably attach this to `List` since you are not calling any of the MutableList interfaces?

Reply:
Good catch !!!